src/application/F3DOptionsParser.cxx

```diff
@@ -528,57 +533,65 @@ std::string ConfigurationOptions::GetSystemSettingsDirectory()
 }
 
 //----------------------------------------------------------------------------
-std::string ConfigurationOptions::GetBinarySettingsDirectory()
+std::string ConfigurationOptions::GetBinaryConfigFileDirectory()
 {
-  std::string directoryPath = "";
+  std::filesystem::path dirPath;
   std::string errorMsg, programFilePath;
-  if (vtksys::SystemTools::FindProgramPath(this->Argv[0], programFilePath, errorMsg))
+  try
   {
-    // resolve symlinks
-    programFilePath = vtksys::SystemTools::GetRealPath(programFilePath);
-    directoryPath = vtksys::SystemTools::GetProgramPath(programFilePath);
-    std::string separator;
-#if defined(_WIN32)
-    separator = "\\";
-    if (directoryPath[directoryPath.size() - 1] != separator[0])
-    {
-      directoryPath.append(separator);
-    }
-#else
-    separator = "/";
-    directoryPath += separator;
-#endif
-    directoryPath += "..";
+    // TODO may not work on windows ?
+    dirPath = std::filesystem::canonical(std::filesystem::path(this->Argv[0]))
+                .parent_path()
+                .parent_path()
+                .string();
 #if F3D_MACOS_BUNDLE
-    if (vtksys::SystemTools::FileExists(directoryPath + "/Resources"))
+    for (auto const& dirEntry : std::filesystem::directory_iterator{ dirPath })
     {
-      directoryPath += "/Resources";
+      if (dirEntry.path().filename() == "Resources")
+      {
+        dirPath = dirEntry.path();
+      }
     }
 #endif
-    directoryPath = vtksys::SystemTools::CollapseFullPath(directoryPath);
-    directoryPath += separator;
   }
-  return directoryPath;
+  catch (const std::filesystem::filesystem_error&)
+  {
+    f3d::log::error("Cannot recover binary config file directory: ", dirPath.string());
```

Comment:
calling `e.what()` could be useful?

Reply:
not needed after all, this exception is excpected in that specific case.